command.go

```diff
@@ -434,7 +437,11 @@ func (c *Command) UsageFunc() (f func(*Command) error) {
 	}
 	return func(c *Command) error {
 		c.mergePersistentFlags()
-		err := tmpl(c.OutOrStderr(), c.UsageTemplate(), c)
+		fn := defaultUsageFunc
+		if c.usageTemplate != nil {
+			fn = c.usageTemplate.fn
+		}
```

Comment:
Unlike the previous code which called `UsageTemplate()` which would return the usage template of the first parent that has one, this new code does not check for parents.

I will push a couple of new tests to check this case.
I'll also try to fix it.

Reply:
Doesn't the code above do it? This is the full func:

```
func (c *Command) UsageFunc() (f func(*Command) error) {
	if c.usageFunc != nil {
		return c.usageFunc
	}
	if c.HasParent() {
		return c.Parent().UsageFunc() // <-- this call?
	}
	return func(c *Command) error {
		c.mergePersistentFlags()
		fn := defaultUsageFunc
		if c.usageTemplate != nil {
			fn = c.usageTemplate.fn
		}
		err := fn(c.OutOrStderr(), c)
		if err != nil {
			c.PrintErrln(err)
		}
		return err
	}
}
```

Reply:
Hmm. I see your point. Let me dig a bit.

Reply:
So the fields `usageFunc` and `usageTemplate` are two different ways a program can override the usage text. If a program calls `SetUsageTemplate()` only, it will affect the call to `UsageFunc` but only through the code I’m pointing out. 

What had me confused, and possibly you as well, is that because of the line you point out, it gives the impression that the part calling `usageTemplate` can only work on the root command.  I had to debug to realize the the `c` command in the function is passed as an argument and is **not** the same as the `c` command in the code you point out.  So, in the function we start with a `c` being a sub-command, and we still need to go through the parents looking for a `usageTemplate`

Reply:
I will change the name of the parameter to the function, it is much too confusing.

Reply:
Turns out that it is nice to have this "confusing" shadowing.  It avoids the risk of using the other `c` command inside the function by mistake.  I'll leave it alone.

Reply:
I agree.